pkg/controller/extensions/request/extensionrequest.go

```diff
@@ -0,0 +1,561 @@
+			OwnerReferences: []metav1.OwnerReference{i.OwnerReference()},
+		},
+		Spec: batchv1.JobSpec{
+			BackoffLimit: &jobBackoff,
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					RestartPolicy:      corev1.RestartPolicyNever,
+					ServiceAccountName: extensionManagerName,
+					InitContainers: []corev1.Container{
+						{
+							Name:    "extension-package",
+							Image:   getPackageImage(i.Spec),
+							Command: []string{"cp", "-R", "/.registry/", "/ext-pkg/"},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      packageContentsVolumeName,
+									MountPath: "/ext-pkg",
+								},
+							},
+						},
+					},
+					Containers: []corev1.Container{
+						{
+							Name:  "extension-executor",
+							Image: executorInfo.image,
+							// "--debug" can be added to this list of Args to get debug output from the job,
+							// but note that will be included in the stdout from the pod, which makes it
+							// impossible to create the resources that the job unpacks.
+							Args: []string{"extension", "unpack", "--content-dir=/ext-pkg"},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      packageContentsVolumeName,
+									MountPath: "/ext-pkg",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: packageContentsVolumeName,
+							VolumeSource: corev1.VolumeSource{
+								EmptyDir: &corev1.EmptyDirVolumeSource{},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (jc *extensionRequestJobCompleter) handleJobCompletion(ctx context.Context, i *v1alpha1.ExtensionRequest, job *batchv1.Job) error {
+	extensionRecord := &v1alpha1.Extension{}
+
+	// find the pod associated with the given job
+	podName, err := jc.findPodNameForJob(ctx, job)
+	if err != nil {
+		return err
+	}
+
+	// read full output from job by retrieving the logs for the job's pod
+	b, err := jc.readPodLogs(job.Namespace, podName)
+	if err != nil {
+		return err
+	}
+
+	// decode and process all resources from job output
+	d := yaml.NewYAMLOrJSONDecoder(b, 4096)
+	for {
+		obj := &unstructured.Unstructured{}
+		if err := d.Decode(&obj); err != nil {
+			if err == io.EOF {
+				// we reached the end of the job output
+				break
+			}
+			return fmt.Errorf("failed to parse output from job %s: %+v", job.Name, err)
+		}
+
+		// process and create the object that we just decoded
+		if err := jc.createJobOutputObject(ctx, obj, i, job); err != nil {
+			return err
+		}
+
+		if isExtensionObject(obj) {
+			// we just created the extension record, try to fetch it now so that it can be returned
+			n := types.NamespacedName{Name: obj.GetName(), Namespace: obj.GetNamespace()}
+			if err := jc.kube.Get(ctx, n, extensionRecord); err != nil {
+				return fmt.Errorf("failed to retrieve created extension record %s from job %s: %+v", obj.GetName(), job.Name, err)
+			}
+		}
+	}
+
+	if extensionRecord == nil {
```

Comment:
This condition is always false.

Reply:
good catch, fixing this logic now so that the extension record is only non-nil when we GET it